lib/ide/ide/src/view/view_layout.rs

```diff
@@ -0,0 +1,151 @@
+        self.keyboard_closure.as_ref().map(|keyboard_closure| {
+            let body = document().unwrap().body().unwrap();
+            let callback : &Function = keyboard_closure.as_ref().unchecked_ref();
+            body.remove_event_listener_with_callback("keydown", callback).ok();
+        });
+    }
+}
+
+impl ViewLayoutData {
+    fn set_view_mode(&mut self, view_mode:ViewMode) {
+        self.view_mode = view_mode;
+        self.update();
+    }
+
+    fn switch_mode(&mut self) {
+        if let ViewMode::Half = self.view_mode {
+            self.set_view_mode(ViewMode::Full)
+        } else {
+            self.set_view_mode(ViewMode::Half)
+        }
+    }
+
+    fn set_dimensions(&mut self, dimensions:Vector2<f32>) {
+        self.dimensions = dimensions;
+        self.update();
+    }
+
+    fn update(&mut self) {
+        let dimensions = self.dimensions;
+        let (position,dimensions) = match self.view_mode {
+            ViewMode::Full => {
+                let position   = Vector2::new(0.0, dimensions.y);
+                (position,dimensions)
+            },
+            ViewMode::Half => {
+                let position   = Vector2::new(0.0, dimensions.y / 2.0);
+                let dimensions = Vector2::new(dimensions.x, dimensions.y / 2.0);
+                (position,dimensions)
+            }
+        };
+        self.text_editor.set_dimensions(dimensions);
+        self.text_editor.set_position(position);
+        self.text_editor.update();
+    }
+}
+
+
+
+// ==================
+// === ViewLayout ===
+// ==================
+
+/// Initial implementation of ViewLayout with a single TextEditor. Pressing ctrl+f toggles
+/// fullscreen mode.
+#[derive(Debug,Clone)]
+pub struct ViewLayout {
+    data : Rc<RefCell<ViewLayoutData>>
+}
+
+impl ViewLayout {
+    /// Creates a new ViewLayout with a single TextEditor.
+    pub fn default(world:&World) -> Self {
+        let text_editor      = TextEditor::new(&world);
+        let keyboard_closure = None;
+        let view_mode        = ViewMode::Half;
+        let dimensions       = zero();
+        let data             = ViewLayoutData{text_editor,keyboard_closure,view_mode,dimensions};
+        let data             = Rc::new(RefCell::new(data));
+        Self {data}.init(world)
+    }
+
+    fn init_keyboard(self) -> Self {
+        let data    = Rc::downgrade(&self.data);
+        let closure = move |event:KeyboardEvent| {
+            const F_KEY : u32 = 70;
+            if event.ctrl_key() && event.key_code() == F_KEY {
+                data.upgrade().map(|data| {
+                    data.borrow_mut().switch_mode()
+                });
+                event.prevent_default();
+            }
+        };
+        let closure : Box<dyn FnMut(KeyboardEvent)> = Box::new(closure);
+        let keyboard_closure                        = Closure::wrap(closure);
+        let callback : &Function                    = keyboard_closure.as_ref().unchecked_ref();
```

Comment:
Same pattern as in `text_editor.rs` -- please think if there's something we could refactor and use for this boxed closure setup.

Reply:
Oh, this is very temporary. I will refactor it to use our keyboard manager in the future